pkg/proxy/iptables/proxier.go

```diff
@@ -134,6 +134,31 @@ func (lkct LinuxKernelCompatTester) IsCompatible() error {
 const sysctlRouteLocalnet = "net/ipv4/conf/all/route_localnet"
 const sysctlBridgeCallIPTables = "net/bridge/bridge-nf-call-iptables"
 
+// Since converting probabilities (floats) to strings is expensive
+// and we are using only probabilities in the for of 1/n, we are
+// precomputing some number of them and reuse them in the code belew.
+const numberOfPrecomputed = 1000
+
+var precomputedProbabilities []string
+
+func init() {
+	precomputedProbabilities = make([]string, numberOfPrecomputed)
+	for i := 1; i < numberOfPrecomputed; i++ {
+		precomputedProbabilities[i] = computeProbability(i)
+	}
+}
+
+func computeProbability(n int) string {
+	return fmt.Sprintf("%0.5f", 1.0/float64(n))
+}
+
+func probability(n int) string {
+	if n < numberOfPrecomputed {
+		return precomputedProbabilities[n]
+	}
+	return computeProbability(n)
```

Comment:
If this triggers it's going to trigger periodically.  Why not a map[int]string, and cache results?